src/kernel/schedule.c

```diff
@@ -134,6 +134,12 @@ NOTRACE void __attribute__((noreturn)) runloop_internal()
     cpuinfo ci = current_cpu();
     thunk t;
 
+    if (ci->current_thread && ci->current_thread != INVALID_ADDRESS) {
+        nanos_thread nt = ci->current_thread;
+        if (nt->pause)
```

Comment:
init_closure can't fail (it just initializes a closure struct and doesn't allocate), and frames are initialized to zero. I think this check is to ignore frames which don't have a pause method, so I think a check for null here is valid.

Reply:
Another good catch, thanks! Since the convention seems to be INVALID_ADDRESS over null pointers, it's okay to assume that other code will initialize a field properly if not using it (e.g. other OS code not using pause would need to initialize to INVALID_ADDRESS rather than leave null).

Reply:
Sorry...pause isn't a part of the frame. Currently this will always pass as a valid thread must be a unix thread, and pause is always filled in. Either value is fine I guess.

Reply:
Oops, I got confused, indeed init_closure() can't fail, so nt->pause is always valid. Sorry for the noise